pyiceberg/catalog/rest.py

```diff
@@ -379,17 +382,18 @@ def _fetch_config(self) -> None:
     def _identifier_to_validated_tuple(self, identifier: Union[str, Identifier]) -> Identifier:
         identifier_tuple = self.identifier_to_tuple(identifier)
         if len(identifier_tuple) <= 1:
-            raise NoSuchTableError(f"Missing namespace or invalid identifier: {'.'.join(identifier_tuple)}")
+            raise NoSuchIdentifierError(f"Missing namespace or invalid identifier: {'.'.join(identifier_tuple)}")
         return identifier_tuple
 
-    def _split_identifier_for_path(self, identifier: Union[str, Identifier, TableIdentifier]) -> Properties:
+    def _split_identifier_for_path(self, identifier: Union[str, Identifier, TableIdentifier], kind: str = "table") -> Properties:
         if isinstance(identifier, TableIdentifier):
             if identifier.namespace.root[0] == self.name:
                 return {"namespace": NAMESPACE_SEPARATOR.join(identifier.namespace.root[1:]), "table": identifier.name}
             else:
                 return {"namespace": NAMESPACE_SEPARATOR.join(identifier.namespace.root), "table": identifier.name}
```

Comment:
The [view identifier is also of TableIdentifier](https://github.com/apache/iceberg/blob/6c7964002b097b933053d8f6db63c0851a55a39a/open-api/rest-catalog-open-api.yaml#L2886C39-L2886C54) type, so I think this would be safer, and future proof (for when we support loading views):

```suggestion
                return {"namespace": NAMESPACE_SEPARATOR.join(identifier.namespace.root[1:]), kind: identifier.name}
            else:
                return {"namespace": NAMESPACE_SEPARATOR.join(identifier.namespace.root), kind: identifier.name}
```